src/utils/retries.ts

```diff
@@ -88,4 +106,24 @@ export class RetryOnServerFailure<T, A extends any[]> {
     // Ensure delay is not negative or greater than maxDelay
     return Math.min(maxDelay, Math.max(0, delay));
   };
+
+  private mapErrorIfNeeded(error: any): any {
+    if (error?.status) {
+      return mapHttpStatusError(error);
+    }
+    return error; // Return original error if no mapping is needed
```

Comment:
Do we know what errors we're seeing that end up without an associated `error.status`?

Reply:
I can't quite remember off the top of my head, and definitely need to look into this in the future at some point, but basically something about the BasePineconeError class sometimes has `status` in the json obj that's printed out in the console (if you print the error), but that shows up as `undefined` when you do `error.status`.